lib/active_stash/index_dsl.rb

```diff
@@ -0,0 +1,285 @@
+module ActiveStash
+  # Implements the DSL for adding searchable encrypted indexes to ActiveRecord
+  # models that include ActiveStash::Search.
+  #
+  #   stash_index do
+  #     auto :email, :first_name, :last_name
+  #     unique :email
+  #     range :created_at, :updated_at
+  #     exact :gender
+  #
+  #     match :description, filter_term_bits: 512
+  #
+  #     match_all :first_name, :last_name, :email
+  #
+  #     index_assoc :patient do
+  #       auto :email, :first_name, :last_name
+  #     end
+  #   end
```

Comment:
Should this doc be in the main ActiveStash::Search module instead? (Or as well as?)

Reply:
Yes it should. I will follow up with a new PR to address documentation.